src/main/java/com/minecolonies/api/colony/requestsystem/requestable/deliveryman/Pickup.java

```diff
@@ -5,58 +5,137 @@
 import com.minecolonies.api.util.ReflectionUtils;
 import com.minecolonies.api.util.constant.TypeConstants;
 import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.Tag;
 import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.world.item.ItemStack;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
-import java.util.Set;
+import java.util.*;
 import java.util.stream.Collectors;
 
 /**
  * Class used to represent pickups inside the request system. This class can be used to request a pickup of
  */
 public class Pickup extends AbstractDeliverymanRequestable
 {
+    ////// --------------------------- NBTConstants --------------------------- \\\\\\
+    protected static final String NBT_PICKUP_FILTER = "Filter";
+    ////// --------------------------- NBTConstants --------------------------- \\\\\\
+
     /**
      * Set of type tokens belonging to this class.
      */
     private final static Set<TypeToken<?>>
       TYPE_TOKENS = ReflectionUtils.getSuperClasses(TypeToken.of(Pickup.class)).stream().filter(type -> !type.equals(TypeConstants.OBJECT)).collect(Collectors.toSet());
 
+    /**
+     * The itemstack to filter for this specific pickup.
+     */
+    @Nullable
+    private List<ItemStack> pickupFilter;
```

Comment:
We should just initiate this with an empty list to simplify the rest of the logic.